### PR TITLE
Add rudimentary proto target

### PR DESCRIPTION
**Pull request by oldergod**
will follow up with emitting rules application + exclusivity logic.

### CODE REVIEW

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt [hunk 1]
```diff
@@ -257,6 +260,55 @@ data class KotlinTarget(
...
+                writer.write(protoFile.toSchema())
+              }
+        } catch (e: IOException) {
+          throw IOException("Error emitting $outputFilePath to $outDirectory", e)
+        }
+      }
+    }
+  }
+
+  private fun ProtoFile.isEmpty() = types.isEmpty() && services.isEmpty() && extendList.isEmpty()
```
Comment (swankjesse) [inline wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: neat

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt [hunk 2]
```diff
@@ -257,6 +260,55 @@ data class KotlinTarget(
...
+        val outputDirectory = fs.getPath(outDirectory, relativePath)
+
+        require(Files.notExists(outputDirectory) || Files.isDirectory(outputDirectory)) {
+          "path $outputDirectory exists but is not a directory."
+        }
+        Files.createDirectories(outputDirectory)
+
+        val outputFilePath = outputDirectory.resolve("${protoFile.name()}.proto")
+        try {
+          OutputStreamWriter(Files.newOutputStream(outputFilePath), StandardCharsets.UTF_8)
```
Comment (swankjesse) [inline wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: I wonder if you can use Okio here

File: wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt [hunk 3]
```diff
@@ -151,6 +152,264 @@ class WireRunTest {
...
+    assertThat(fs.find("generated")).containsExactly(
+        "generated/proto/squareup/colors/blue.proto",
+        "generated/proto/squareup/colors/red.proto")
+    assertThat(fs.get("generated/proto/squareup/colors/blue.proto"))
+        .contains("message Blue {")
+    assertThat(fs.get("generated/proto/squareup/colors/red.proto"))
+        .contains("message Red {")
+  }
+
+  @Test @Ignore("To be implemented")
```
Comment (swankjesse) [inline wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt]: Nuke these tests and fix Target to specify the emit types only when they make sense

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt (hunk 2 above)
Comment (oldergod) [reply wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: oh yeah

File: wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt (hunk 3 above)
Comment (oldergod) [reply wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt]: nuked

Comment (oldergod) [reply wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt]: will refactor the target in followup

**Pull request merged**